aten/src/ATen/native/quantized/cpu/fbgemm_utils.cpp

```diff
@@ -212,99 +213,17 @@ Tensor ConvertToChannelsLast3dTensor(const Tensor& src) {
 
 template <int kSpatialDim = 2>
 CAFFE2_API torch::class_<ConvPackedParamsBase<kSpatialDim>> register_conv_params() {
-  using SerializationType = std::tuple<
-    at::Tensor,
-    c10::optional<at::Tensor>,
-    // these are meant to be torch::List<int64_t> but
-    // it's not supported by onnx, so we'll use Tensor as
-    // a workaround
-    torch::List<at::Tensor>,
-    torch::List<at::Tensor>,
-    torch::List<at::Tensor>,
-    at::Tensor>;
   static auto register_conv_params =
     torch::class_<ConvPackedParamsBase<kSpatialDim>>(
         "quantized", "Conv" + c10::to_string(kSpatialDim) + "dPackedParamsBase")
     .def_pickle(
         [](const c10::intrusive_ptr<ConvPackedParamsBase<kSpatialDim>>& params)
         -> SerializationType { // __getstate__
-          at::Tensor weight;
-          c10::optional<at::Tensor> bias;
-          std::tie(weight, bias) = params->unpack();
-          torch::List<at::Tensor> stride;
-          torch::List<at::Tensor> padding;
-          torch::List<at::Tensor> dilation;
-          at::Tensor groups;
-          for (int64_t s : params->stride()) {
-            stride.emplace_back(at::tensor(s));
-          }
-          for (int64_t p : params->padding()) {
-            padding.emplace_back(at::tensor(p));
-          }
-          for (int64_t d : params->dilation()) {
-            dilation.emplace_back(at::tensor(d));
-          }
-          groups = at::tensor(params->groups());
-          return std::make_tuple(
-              std::move(weight),
-              std::move(bias),
-              stride,
-              padding,
-              dilation,
-              groups);
+          return serialize_conv<kSpatialDim>(params);
         },
         [](SerializationType state)
         -> c10::intrusive_ptr<ConvPackedParamsBase<kSpatialDim>> { // __setstate__
```

Comment:
should this function be the place where BC is guaranteed for non-ONXX?  I.e. it would know how to read all past supported versions of the data from disk and convert them to the current version.  Or, does that happen elsewhere?

Reply:
THat's the part that I cannot figure out how to implement